lib/include/openamp/rpmsg_virtio.h

```diff
@@ -143,22 +144,37 @@ void rpmsg_deinit_vdev(struct rpmsg_virtio_device *rvdev);
  *
  * RPMsg virtio has default shared buffers pool implementation.
  * The memory assigned to this pool will be dedicated to the RPMsg
- * virtio. If you prefer to have other shared buffers allocation,
- * you can implement your rpmsg virtio shared buffers pool.
+ * virtio. This function has to be called before calling rpmsg_init_vdev,
+ * to initialize the rpmsg_virtio_shm_pool structure.
  *
- * @param shpool - pointer to the shared buffers pool
+ * @param shpool - pointer to the shared buffers pool structure
  * @param shbuf - pointer to the beginning of shared buffers
  * @param size - shared buffers total size
  */
 void rpmsg_virtio_init_shm_pool(struct rpmsg_virtio_shm_pool *shpool,
-				void *shb, size_t size);
+				void *shbuf, size_t size);
 
 static inline struct rpmsg_device *
 rpmsg_virtio_get_rpmsg_device(struct rpmsg_virtio_device *rvdev)
 {
 	return &rvdev->rdev;
 }
 
+/**
+ * rpmsg_virtio_shm_pool_get_buffer - get buffer in the shared memory pool
+ *
+ * RPMsg virtio has default shared buffers pool implementation.
+ * The memory assigned to this pool will be dedicated to the RPMsg
+ * virtio. If you prefer to have other shared buffers allocation,
+ * you can implement your rpmsg_virtio_shm_pool_get_buffer function.
+ *
```

Comment:
We should add something here like:

If you implement your own rpmsg_virtio_shm_pool_get_buffer the expectation is that one would embedded struct rpmsg_virtio_shm_pool object inside your own struct and utilize something like metal_container_of() to reference back to your own struct in your implementation of rpmsg_virtio_shm_pool_get_buffer